projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -193,7 +193,7 @@ public function add_new_site_link() {
 	 * Adds site card component.
 	 */
 	public function add_site_card_menu() {
-		$default        = 'data:image/svg+xml,' . rawurlencode( '<svg class="gridicon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><title>Globe</title><rect fill-opacity="0" x="0" width="24" height="24"/><g><path fill="#fff" d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm0 18l2-2 1-1v-2h-2v-1l-1-1H9v3l2 2v1.93c-3.94-.494-7-3.858-7-7.93l1 1h2v-2h2l3-3V6h-2L9 5v-.41C9.927 4.21 10.94 4 12 4s2.073.212 3 .59V6l-1 1v2l1 1 3.13-3.13c.752.897 1.304 1.964 1.606 3.13H18l-2 2v2l1 1h2l.286.286C18.03 18.06 15.24 20 12 20z"/></g></svg>' );
+		$default        = plugins_url( 'globe-icon.svg', __FILE__ );
```

Comment:
IIRC the SVG was added inline to avoid an additional file request. Is there a way to avoid the painting aspect of the SVG with a style attribute or something instead?

Reply:
Yea, fair enough. I explored this a bit. It looks like we could address it here: https://github.com/WordPress/WordPress/blob/master/wp-admin/js/svg-painter.js#L189-L193

In short, I think that we'd get the current color and then add a prop like `data-wp-ui-svg-{$COLOR}=none`. 

I don't feel too strongly about the extra request as:

- The SVG should be cached after the initial request
- The dimensions of the image are set to minimize reflow

It's also less likely to break if the SVG handling changes. For example, I imagine that the site icon worked at some point and then something changed to where the SVG had to be base64 encoded.

Reply:
> IIRC the SVG was added inline to avoid an additional file request. Is there a way to avoid the painting aspect of the SVG with a style attribute or something instead?

Maybe it was https://core.trac.wordpress.org/changeset/53050 ?

Here's a bug report for what we're experiencing: https://core.trac.wordpress.org/ticket/55543

Reply:
> Maybe it was https://core.trac.wordpress.org/changeset/53050 ?

Oh, I scrolled far enough to see @obenland's comment https://github.com/Automattic/jetpack/pull/27511#issuecomment-1322511820

Reply:
I could be wrong, but I remember us favoring not bundling SVGs and instead loading them as external (sprites if possible) for better performance. SVGs quickly add up to JS bundles whereas they can just be loaded and cached for a long time as images. @sgomes worked on these before and can probably provide more context.

See https://github.com/Automattic/wp-calypso/pull/32750

Reply:
@simison As far as I can tell, the previous code was embedding the SVG in the HTML, as a data URL, rather than in JS. While still less than ideal (for caching reasons, for example), excess HTML is much cheaper than excess JS for the browser to handle, and should be absolutely fine for such a small amount of code.